core/src/main/java/org/mule/runtime/core/routing/requestreply/AbstractReplyToPropertyRequestReplyReplier.java

```diff
@@ -45,7 +46,9 @@ protected Event processReplyTo(Event event, Event result, ReplyToHandler replyTo
       throws MuleException {
     if (result != null && replyToHandler != null) {
       String requestor = ((InternalMessage) result.getMessage()).getOutboundProperty(MULE_REPLY_TO_REQUESTOR_PROPERTY);
-      if ((requestor != null && !requestor.equals(flowConstruct.getName())) || requestor == null) {
+      if ((requestor != null
+          && !requestor.equals(getLocation().getParts().get(0).getPartPath()))
```

Comment:
Refactor repeated use of this type of thing into static helpers in ComponentLocation.